### PR TITLE
Added config to prevent unobtainable Pokemon from being shiny

### PR DESCRIPTION
**Pull request by fdeblasio**
## Description
Added P_ONLY_OBTAINABLE_SHINIES config to prevent unobtainable Pokemon (pre-Poke Balls and in Battle Pyramid) from being shiny

## **Discord contact info**
Frankfurter0

### CODE REVIEW

**Comment by Sneed69** [comment]
Seems like this would cause your starter to never be shiny also.

**Comment by fdeblasio** [comment]
> Seems like this would cause your starter to never be shiny also.

I think the starter gets taken care of by ScriptGiveMonParameterized in script_pokemon_util.c

File: include/config/pokemon.h [hunk 1]
```diff
@@ -46,6 +46,7 @@
 #define P_SHOW_TERA_TYPE            GEN_8       // Since Gen 9, the Tera Type is shown on the summary screen.
 #define P_TM_LITERACY               GEN_LATEST  // Since Gen 6, TM illiterate Pokémon can learn TMs that teach moves that are in their level-up learnsets.
 #define P_EGG_CYCLE_LENGTH          GEN_LATEST  // Since Gen 8, egg cycles take half as many steps as before.
+#define P_ONLY_OBTAINABLE_SHINIES   FALSE       //If TRUE, Pokémon encountered when FLAG_ADVENTURE_STARTED is not set OR in the Battle Pyramid won't be shiny.
```
Comment (Bassoonian) [inline include/config/pokemon.h:49]:
```suggestion
#define P_ONLY_OBTAINABLE_SHINIES   FALSE       // If TRUE, Pokémon encountered when FLAG_ADVENTURE_STARTED is not set OR in the Battle Pyramid won't be shiny.
```

**Review by Bassoonian — COMMENTED** [review]
This is out of scope, but I feel like a logical analogy (one that will probably get activated a lot) is a separate config that disallows wild shiny Pokémon if your Poké Ball pocket is empty, even halfway the game. Up to you if you'd like to include it in this PR or not :)

**Comment by fdeblasio** [comment]
> This is out of scope, but I feel like a logical analogy (one that will probably get activated a lot) is a separate config that disallows wild shiny Pokémon if your Poké Ball pocket is empty, even halfway the game. Up to you if you'd like to include it in this PR or not :)

I like this idea! If I can figure out how to make it happen, I'll add it to the PR

**Comment by fdeblasio** [comment]
> This is out of scope, but I feel like a logical analogy (one that will probably get activated a lot) is a separate config that disallows wild shiny Pokémon if your Poké Ball pocket is empty, even halfway the game. Up to you if you'd like to include it in this PR or not :)

I added the functionality, but I'm not thrilled with the name of the config. If anyone has any better names, I'd love some recommendations

**Comment by AsparagusEduardo** [comment]
> Alternatively, you could make it a flag define and the user would just use any flag for their specific game
> 
> this is not blocking feedback

Tbh, I disagree. Flags are a lot of the time modified by the user, meaning that `FLAG_ADVENTURE_STARTED` might be gone in downstream hacks. I recommend implementing this as a flag define instead.

**Review by AlexOn1ine — APPROVED** [review]
I'm going to merge this. @AsparagusEduardo Please make an issue if you would like any follow up on this.

**Pull request merged**